### PR TITLE
[MAINT] sort entries in docs.py in alphabetical order

### LINKED ISSUE #3622: sort entries in docs.py in alphabetical order
If there is no objection I would like in another PR to sort the entries in this file alphabetically: this may help navigate it.

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/3621#discussion_r1147498458_

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3622 

Changes proposed in this pull request:

- sort entries in docs.py in alphabetical order

### CODE REVIEW

File: nilearn/_utils/docs.py [hunk 1]
```diff
@@ -3,310 +3,195 @@
...
-          If True, it is equivalent to ``opening=1``.
-        - An integer `n`: The :term:`opening<Opening>` is performed via `n`
-          :term:`erosions<Erosion>` (see :func:`scipy.ndimage.binary_erosion`).
-          The largest connected component is then estimated if ``connected`` is
-          set to True, and 2`n` :term:`dilation<Dilation>` operations are
-          performed (see :func:`scipy.ndimage.binary_dilation`) followed by
-          `n` :term:`erosions<Erosion>`. This corresponds to 1
-          :term:`opening<Opening>` operation of order `n` followed by a
-          :term:`closing<Closing>` operator of order `n`.
+# classifier_options
```
Comment (ymzayek) [inline nilearn/_utils/docs.py:125]: You can use intersphinx linking here and get rid of `base_url` and `SKLEARN_LINKS`. See example below

File: nilearn/_utils/docs.py [hunk 2]
```diff
@@ -3,310 +3,195 @@
...
+logistic = "Logistic regression"
+rc = "Ridge classifier"
+dc = "Dummy classifier with stratified strategy"
 
-    .. note::
+docdict['classifier_options'] = f"""
 
-        Turning off :term:`opening<Opening>` (``opening=False``) will also
-        prevent any smoothing applied to the image during the mask computation.
+        - `svc`: `{svc} <%(svc)s>`_ with L2 penalty.
```
Comment (ymzayek) [inline nilearn/_utils/docs.py:133]: Example:

```suggestion
        - `svc`: :class:`{svc} <sklearn.svm.SVC>` with L2 penalty.
```

**Review by ymzayek — APPROVED** [review]
LGTM! Feel free to do the suggestion in another PR but it's also quick to do

File: nilearn/_utils/docs.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/_utils/docs.py:115]: should be done

Comment (ymzayek) [reply nilearn/_utils/docs.py:115]: Great thx! Looks right and links work in the doc artifacts

Comment (Remi-Gau) [reply nilearn/_utils/docs.py:115]: will wait for CI to complete and will merge then

**Comment by Remi-Gau** [comment]
merging
thanks for the @ymzayek @NicolasGensollen

**Pull request merged**